src/Umbraco.Infrastructure/PropertyEditors/MediaPicker3PropertyEditor.cs

```diff
@@ -383,5 +391,101 @@ public IEnumerable<ValidationResult> Validate(
+                PropertyValidationContext validationContext)
+            {
+                if (value is null || configuration is null)
+                {
+                    return [];
+                }
+
+                var allowedTypes = configuration.Filter?.Split(Constants.CharArrays.Comma, StringSplitOptions.RemoveEmptyEntries);
+
+                // No allowed types = all types are allowed
+                if (allowedTypes is null || allowedTypes.Length == 0)
+                {
+                    return [];
+                }
+
+                foreach (MediaWithCropsDto media in value)
+                {
+                    IMediaType? type = _mediaTypeService.Get(media.MediaTypeAlias);
+
+                    if (type is null || allowedTypes.Contains(type.Key.ToString()) is false)
+                    {
+                        return
+                        [
+                            new ValidationResult(
+                                _localizedTextService.Localize("validation", "invalidMediaType"),
+                                new[] { "invalidMediaType" })
+                        ];
+                    }
+                }
+
+                return [];
+            }
+        }
+
+        internal class StartNodeValidator : ITypedJsonValidator<List<MediaWithCropsDto>, MediaPicker3Configuration>
+        {
+            private readonly ILocalizedTextService _localizedTextService;
+            private readonly IMediaNavigationQueryService _mediaNavigationQueryService;
+
+            public StartNodeValidator(
+                ILocalizedTextService localizedTextService,
+                IMediaNavigationQueryService mediaNavigationQueryService)
+            {
+                _localizedTextService = localizedTextService;
+                _mediaNavigationQueryService = mediaNavigationQueryService;
+            }
+
+            public IEnumerable<ValidationResult> Validate(
+                List<MediaWithCropsDto>? value,
+                MediaPicker3Configuration? configuration,
+                string? valueType,
+                PropertyValidationContext validationContext)
+            {
+                if (value is null || configuration?.StartNodeId is null)
+                {
+                    return [];
+                }
+
+                foreach (MediaWithCropsDto media in value)
+                {
+                    if (_mediaNavigationQueryService.TryGetAncestorsKeys(media.MediaKey, out IEnumerable<Guid> ancestorsKeys) is false)
+                    {
+                        continue;
+                    }
+
+                    if (ancestorsKeys.Contains(configuration.StartNodeId.Value) is false)
+                    {
+                        return
+                        [
+                            new ValidationResult(
+                                _localizedTextService.Localize("validation", "invalidStartNode"),
+                                new[] { "invalidStartNode" })
```

Comment:
Not sure what we should be providing here as this `memberNames` parameter.  I assume in a typical form post it would be the name of the field that's invalid, so perhaps in this case it should just be `value`?  At least that's what I saw the simpler property editor's validation was returning.

Reply:
I've changed them to value :+1: Not sure what we use them for, just found it nice for testing to assert that the "correct" validator failed :smile: